[bug] Unable to detect if file flag was actually provided?

Hi There.

This has been my go-to library for argument parsing, and I recently tried to use this to parse file arguments. However, it seems impossible to determine if a flag was actually provided. For example:
```go
configFile := parser.File("c", "config", etc)
parser.Parse()
if configFile != nil {
    // Do stuff with the file
}
// Do stuff if -c wasn't provided
```
This does not work, because the configFile is always non-nil, as the parser seems to work by overwriting the internal structures of os.File, and if the argument isn't provided, the configFile will still be non-nil. Furthermore, the standard library doesn't help in this regard, since all the regular functions assume that the file object is valid, so if I try something like `configFile.Name()`, it just panics.

Is there something I'm missing here, or is there no way to distinguish if the -c flag was provided, without reflection nonsense or panic handling?

Basically, the library fails to cater for scenarios when the file flag is optional. Here's an example:

```go
package main

import (
	"log"
	"os"

	"github.com/akamensky/argparse"
)

func main() {
	parser := argparse.NewParser("example", "example")

	// Note this is an optional file. There's no default, and it's not required
	configFile := parser.File("c", "config", os.O_RDONLY, os.FileMode(0), &argparse.Options{
		Help: "config file",
	})

	err := parser.Parse(os.Args)
	if err != nil {
		log.Fatal(err)
	}

	if configFile != nil {
		// Do something if the config file does exist
		log.Println(configFile.Name())
	} else {
		// Do something if the -c flag isn't provided
		log.Println("no config file")
	}
}
```

And here's what I see if I try run that:
```
ryanpersonal@Ryans-MacBook-Pro ~/D/playground> go run . -c invalidfile
2021/08/12 18:22:10 open invalidfile: no such file or directory
exit status 1
# This is expected, since the library checks if the file exists, and returns an error otherwise

ryanpersonal@Ryans-MacBook-Pro ~/D/playground [1]> go run . -c main.go
2021/08/12 18:22:14 main.go
# This is expected, since the library will open the file if it exists

ryanpersonal@Ryans-MacBook-Pro ~/D/playground> go run .
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x10b95a8]

goroutine 1 [running]:
os.(*File).Name(...)
	/usr/local/go/src/os/file.go:55
main.main()
	/Users/ryanpersonal/Documents/playground/main.go:25 +0x168
exit status 2
# This is unexpected, since there should be a way to gracefully handle a case where the user hasn't provided a file, since it's an optional argument.
```

I can understand that changing the nil-ness of the File pointer isn't possible after it's been returned, but it should then instead either return a double pointer (I don't recommend this since it would be a breaking change), or return a pointer to a second boolean value to state if the parameter was actually provided (this is still breaking, but it's much easier to retrofit code to work with this by adding a `, _`, and this way keeps consistency with the other parameter types)

--- a/argparse.go
+++ b/argparse.go
@@ -329,8 +329,9 @@ func (o *Command) File(short string, long string, flag int, perm os.FileMode, op
 }
 
 // List creates new list argument. This is the argument that is allowed to be present multiple times on CLI.
-// All appearances of this argument on CLI will be collected into the list of default type values ​​which is strings. If no argument
-// provided, then the list is empty. Takes same parameters as String
+// All appearances of this argument on CLI will be collected into the list of default type values which is strings.
+// If no argument provided, then the list is empty.
+// Takes same parameters as String.
 // Returns a pointer the list of strings.
 func (o *Command) List(short string, long string, opts *Options) *[]string {
 	return o.StringList(short, long, opts)

--- a/misc.go
+++ b/misc.go
@@ -1,0 +1,13 @@
+package argparse
+
+import (
+	"os"
+	"reflect"
+)
+
+// IsNilFile allows to test whether returned `*os.File` has been initialized with data passed on CLI.
+// Returns true if `fd == &{nil}`, which means `*os.File` was not initialized, false if `fd` is
+// a fully initialized `*os.File` or if `fd == nil`.
+func IsNilFile(fd *os.File) bool {
+	return reflect.DeepEqual(fd, &os.File{})
+}

--- a/misc_test.go
+++ b/misc_test.go
@@ -1,0 +1,30 @@
+package argparse
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func TestIsNilFile(t *testing.T) {
+	var b bool
+	b = IsNilFile(nil)
+	// nil != &{nil}
+	if b {
+		t.Errorf("Expected [%v], got [%v]", "false", b)
+	}
+
+	var fd os.File
+	b = IsNilFile(&fd)
+	// &{nil} != &{nil}
+	if !b {
+		t.Errorf("Expected [%v], got [%v]", "true", !b)
+	}
+
+	fdp, _ := ioutil.TempFile(os.TempDir(), "test")
+	b = IsNilFile(fdp)
+	// *os.File != &{nil}
+	if b {
+		t.Errorf("Expected [%v], got [%v]", "false", !b)
+	}
+}

